File: CS/DxSample/Editors/CheckedListBoxExControlHandler.cs
using DevExpress.XtraEditors;
using System.Drawing;
using DevExpress.XtraEditors.ViewInfo;

namespace DxSample.Editors {
    public class CheckedListBoxExControlHandler : CheckedListBoxControlHandler {
        public CheckedListBoxExControlHandler(CheckedListBoxExControl control) : base(control) { }

        public new CheckedListBoxExControl OwnerControl {
            get { return (CheckedListBoxExControl)base.OwnerControl; }
        }

        protected override ListBoxControlHandler.ListBoxControlState CreateState(ListBoxControlHandler.HandlerState state) {
            if (state == HandlerState.Default || state == HandlerState.SingleSelect)
                return new CheckedListBoxExControlHandler.CheckedSingleSelectState(this);
            return base.CreateState(state);
        }

        private new class CheckedSingleSelectState : CheckedListBoxControlHandler.CheckedSingleSelectState {
            public CheckedSingleSelectState(CheckedListBoxExControlHandler handler) : base(handler) { }

            protected new CheckedListBoxExControl ListBox {
                get { return (CheckedListBoxExControl)base.ListBox; }
            }

            protected override bool CanChangePressState(Point location, int selIndex, bool checkOnClick) {
                bool canChangePressState = true;
                if (!this.ListBox.UseWholeItemToChangeCheckedStateByMouse) {
                    CheckedListBoxViewInfo.CheckedItemInfo item = this.ViewInfo.GetItemByIndex(this.ViewInfo.HotItemIndex);
                    canChangePressState = item.CheckArgs.Bounds.Contains(location);
                }
                return canChangePressState && base.CanChangePressState(location, selIndex, checkOnClick);
            }
        }
    }
}

File: VB/DxSample/Editors/CheckedListBoxExControlHandler.vb
Imports DevExpress.XtraEditors
Imports System.Drawing
Imports DevExpress.XtraEditors.ViewInfo

Namespace DxSample.Editors
    Public Class CheckedListBoxExControlHandler
        Inherits CheckedListBoxControlHandler

        Public Sub New(ByVal control As CheckedListBoxExControl)
            MyBase.New(control)
        End Sub

        Public Shadows ReadOnly Property OwnerControl() As CheckedListBoxExControl
            Get
                Return CType(MyBase.OwnerControl, CheckedListBoxExControl)
            End Get
        End Property

        Protected Overrides Function CreateState(ByVal state As ListBoxControlHandler.HandlerState) As ListBoxControlHandler.ListBoxControlState
            If state = HandlerState.Default OrElse state = HandlerState.SingleSelect Then
                Return New CheckedListBoxExControlHandler.CheckedSingleSelectState(Me)
            End If
            Return MyBase.CreateState(state)
        End Function

        Private Shadows Class CheckedSingleSelectState
            Inherits CheckedListBoxControlHandler.CheckedSingleSelectState

            Public Sub New(ByVal handler As CheckedListBoxExControlHandler)
                MyBase.New(handler)
            End Sub

            Protected Shadows ReadOnly Property ListBox() As CheckedListBoxExControl
                Get
                    Return CType(MyBase.ListBox, CheckedListBoxExControl)
                End Get
            End Property

            Protected Overrides Function CanChangePressState(ByVal location As Point, ByVal selIndex As Integer, ByVal checkOnClick As Boolean) As Boolean

                Dim canChangePressState_Renamed As Boolean = True
                If Not Me.ListBox.UseWholeItemToChangeCheckedStateByMouse Then
                    Dim item As CheckedListBoxViewInfo.CheckedItemInfo = Me.ViewInfo.GetItemByIndex(Me.ViewInfo.HotItemIndex)
                    canChangePressState_Renamed = item.CheckArgs.Bounds.Contains(location)
                End If
                Return canChangePressState_Renamed AndAlso MyBase.CanChangePressState(location, selIndex, checkOnClick)
            End Function
        End Class
    End Class
End Namespace
